x-pack/plugins/ml/public/application/services/anomaly_explorer_charts_service.ts

```diff
@@ -381,8 +384,47 @@ export class AnomalyExplorerChartsService {
     return combinedJobs;
   }
 
+  public loadDataForCharts$(
+    jobIds: string[],
+    earliestMs: number,
+    latestMs: number,
+    influencers: EntityField[] = [],
+    selectedCells: AppStateSelectedCells | undefined,
+    influencersFilterQuery: InfluencersFilterQuery
+  ) {
+    if (
+      selectedCells === undefined &&
+      influencers.length === 0 &&
+      influencersFilterQuery === undefined
+    ) {
+      of([]);
+    }
+
+    return from(
+      this.mlResultsService.getRecordsForInfluencer(
+        jobIds,
+        influencers,
+        0,
+        earliestMs,
+        latestMs,
+        500,
+        influencersFilterQuery
+      )
+    ).pipe(
+      switchMap((resp) => {
+        if (
+          (selectedCells !== undefined && Object.keys(selectedCells).length > 0) ||
+          influencersFilterQuery !== undefined
+        ) {
+          return of(resp.records);
+        }
+
+        return of([]);
+      })
```

Comment:
You don't perform any async operation here, hence you can simply use `map` operator. 
```suggestion
      map((resp) => {
        if (
          (selectedCells !== undefined && Object.keys(selectedCells).length > 0) ||
          influencersFilterQuery !== undefined
        ) {
          return resp.records;
        }

        return [];
      })
```